internal/runners/use/use.go

```diff
@@ -66,14 +69,44 @@ func (u *Use) Run(params *Params) error {
 
 	checker.RunUpdateNotifier(u.svcModel, u.out)
 
-	projectPath, err := u.checkout.Run(params.Namespace, "", "")
+	projectsDir, err := storage.ProjectsDir()
 	if err != nil {
-		return locale.WrapError(err, "err_checkout_project", params.Namespace.String())
+		return locale.WrapError(err, "err_use_cannot_determine_projects_dir", "")
 	}
 
-	proj, err := project.FromPath(projectPath)
-	if err != nil {
-		return locale.WrapError(err, "err_activate_projectfrompath")
+	var projectDir string
+	if params.PreferredPath == "" {
+		projectDir = filepath.Join(projectsDir, params.Namespace.Project)
+	} else {
+		projectDir = params.PreferredPath
+	}
+
+	var proj *project.Project
+
+	if params.Namespace.Owner != "" {
+		logging.Debug("Checking out %s to %s", params.Namespace.String(), projectDir)
+		projectPath, err := u.checkout.Run(params.Namespace, params.Branch, projectDir)
+		if err != nil {
+			return locale.WrapError(err, "err_use_checkout_project", params.Namespace.String())
+		}
+
+		proj, err = project.FromPath(projectPath)
+		if err != nil {
+			return locale.WrapError(err, "err_use_project_frompath")
+		}
+	} else {
+		logging.Debug("Using an already checked out project: %s", projectDir)
+
+		proj, err = project.FromPath(projectDir)
+		if err != nil {
+			wrapped := locale.WrapInputError(err, "err_use_project_not_checked_out", "", params.Namespace.Project, projectDir)
+			errs.AddTips(wrapped, locale.Tl("use_checkout_first", "", params.Namespace.Project))
+			return wrapped
+		}
+
+		if params.Branch != "" && proj.BranchName() != params.Branch {
+			return locale.NewInputError("err_conflicting_branch_while_checkedout", "", params.Branch, proj.BranchName())
+		}
```

Comment:
I _think_ this is missing the following AC:
 - If namespace or projectName matches an already locally checked out project; use that instead of checking it out fresh

Wouldn't we have to check the config to see if a namespace matches a project checked out somewhere else on the system? This would mean that if I activated a project in a directory other than `~/Projects` I should be able `state use` that project without it being checked out again.

Reply:
You are correct that this does not handle the case of a project checked out elsewhere. I have to add that. There is a unit test for this AC for projects checked out without using a custom path: https://github.com/ActiveState/cli/pull/1941/files#diff-5c34fd97485de0785b3179af8a6a7a4d5fe90ec253e26934b92640142105ceadR90